### PR TITLE
Add AI-Powered Hashtag Suggestions for Social Media Posts

### LINKED ISSUE #123: [FEATURE REQUEST] Add "Back to Top" Button to the Website
The website currently lacks a "Back to Top" button, which can improve user experience by allowing users to quickly scroll back to the top of the page. This is especially useful for pages with a lot of content that require extensive scrolling.

Be hidden by default and only appear after a certain scroll threshold (e.g., 200px down the page).
Include smooth scroll behavior to enhance user experience.
Be responsive and compatible with both light and dark themes.

please assign this issue to me @himeshparashar under gssoc-ext and hacktoberfest

### LINKED ISSUE #124: Add AI-Powered Hashtag Suggestions for Social Media Posts
Enhance Social Morph's capabilities by integrating an AI-powered hashtag suggestion feature. This feature should analyze the content of the user's post or the provided keywords and suggest trending and relevant hashtags that can maximize the reach and engagement of the post.

Requirements:

Implement an AI-based algorithm to analyze the content of the post.
Provide a list of relevant hashtags based on current trends and keywords.

@himeshparashar please assign this to me under gssoc-ext and hacktoberfest

### PR DESCRIPTION
**Pull request by sezallagwal**
## 🔨 Changes Made

- Enhance Social Morph's capabilities by integrating an AI-powered hashtag suggestion feature. 
- Implement an AI-based algorithm to analyze the content of the post.
- Provide a list of relevant hashtags based on current trends and keywords.

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings.
- [x] I have tested it locally and it works fine.
- [ ] Any dependent changes have been merged and published in downstream modules.

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [ ] Bug fix (non-breaking change which fixes an issue) 🐛
- [x] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## 🤝 Related Issues

- Fixes #124

## How should this be tested?

- [x] Test A: use topic: "sezal"
- [x] Test B: use topic: "sezal lagwal"

## 📸 Screenshots (if applicable)

image

image

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by sezallagwal** [comment]
@himeshparashar can you add level 3?

File: app/(root)/dashboard/page.tsx [hunk 1]
```diff
@@ -1,11 +1,155 @@
...
 
-import React from "react";
+import React, { useState } from "react";
+import ReactMarkdown from "react-markdown";
 import { Plus } from "lucide-react";
+import { GoogleGenerativeAI } from "@google/generative-ai";
+
+const apiKey = process.env.NEXT_PUBLIC_GEMINI_API_KEY;
+
+console.log(apiKey);
```
Comment (sudhanshutech) [inline app/(root)/dashboard/page.tsx:10]: remove the console please and specially don't console a api key😅

File: app/(root)/dashboard/page.tsx [hunk 2]
```diff
@@ -1,11 +1,155 @@
...
+
+    setIsLoading(true);
+    setError(null);
+    setGeneratedContent("");
+    setGeneratedHashtags([]);
+
+    try {
+      if (!apiKey) {
+        throw new Error(
+          "Gemini API key is not set. Please check your environment variables."
```
Comment (sudhanshutech) [inline app/(root)/dashboard/page.tsx:43]: User can not check a environment variable, instead you can say: Please contact project owner or support

File: app/(root)/dashboard/page.tsx [hunk 3]
```diff
@@ -1,11 +1,155 @@
...
+        );
+      }
+
+      const fullPrompt = `${customPrompt} ${inputText}`;
+
+      console.log("Generating content with prompt:", fullPrompt);
```
Comment (sudhanshutech) [inline app/(root)/dashboard/page.tsx:49]: remove please

File: app/(root)/dashboard/page.tsx [hunk 4]
```diff
@@ -1,11 +1,155 @@
...
+      const model = genAI.getGenerativeModel({ model: "gemini-pro" });
+
+      console.log("Generating content...");
+      const result = await model.generateContent(fullPrompt);
+
+      console.log("Processing response...");
```
Comment (sudhanshutech) [inline app/(root)/dashboard/page.tsx:55]: remove all consoles when you have done testing

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

**Comment by sezallagwal** [comment]
@sudhanshutech is it okay now?

**Pull request merged**